app/src/main/java/co/stonephone/stonecamera/plugins/SettingsTray.kt

```diff
@@ -0,0 +1,171 @@
+                                            Row(
+                                                modifier = Modifier
+                                                    .fillMaxWidth()
+                                                    .padding(horizontal = 12.dp, vertical = 2.dp),
+                                                horizontalArrangement = Arrangement.SpaceBetween
+                                            ) {
+                                                setting.options.map { option ->
+                                                    Box(modifier = Modifier
+                                                        .then(modifier)
+                                                        .clickable {
+                                                            viewModel.setSetting(
+                                                                setting.key, option
+                                                            )
+                                                        }) {
+                                                        setting.render(option, option == value)
+                                                    }
+                                                }
+                                            }
+                                        }
+
+                                        is PluginSetting.ScalarSetting -> {
+                                            Slider(
+                                                value = value as Float,
+                                                onValueChange = { newValue ->
+                                                    viewModel.setSetting(setting.key, newValue)
+                                                },
+                                                valueRange = setting.minValue..setting.maxValue,
+                                                modifier = modifier
+                                            )
+                                        }
+
+                                        is PluginSetting.CustomSetting -> setting.customRender(
+                                            viewModel, value
+                                        ) { value ->
+                                            viewModel.setSetting(setting.key, value)
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(8.dp),
+            horizontalArrangement = Arrangement.End,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            IconButton(
+                onClick = {
+                    showSettingsTray = true
+                },
+                modifier = Modifier
+                    .size(34.dp)
```

Comment:
This and similar values seem ripe for someone wanting to twiddle with them to make the UI work on various screen resolutions and devices, yet they're hard-coded.